x/stake/keeper/key.go

```diff
@@ -84,114 +90,161 @@ func GetValidatorPowerRank(validator types.Validator, pool types.Pool) []byte {
 				append(heightBytes, counterBytes...)...)...)...)
 }
 
-// get the key for the accumulated update validators
+// get the key for the accumulated update validators.
+// VALUE: abci.Validator
+// note records using these keys should never persist between blocks
 func GetTendermintUpdatesKey(ownerAddr sdk.Address) []byte {
 	return append(TendermintUpdatesKey, ownerAddr.Bytes()...)
 }
 
 //________________________________________________________________________________
 
-// get the key for delegator bond with validator
-func GetDelegationKey(delegatorAddr, validatorAddr sdk.Address, cdc *wire.Codec) []byte {
-	return append(GetDelegationsKey(delegatorAddr, cdc), validatorAddr.Bytes()...)
+// get the key for delegator bond with validator.
+// VALUE: stake/types.Delegation
+func GetDelegationKey(delegatorAddr, validatorAddr sdk.Address) []byte {
+	return append(GetDelegationsKey(delegatorAddr), validatorAddr.Bytes()...)
 }
 
 // get the prefix for a delegator for all validators
-func GetDelegationsKey(delegatorAddr sdk.Address, cdc *wire.Codec) []byte {
-	res := cdc.MustMarshalBinary(&delegatorAddr)
-	return append(DelegationKey, res...)
+func GetDelegationsKey(delegatorAddr sdk.Address) []byte {
+	return append(DelegationKey, delegatorAddr.Bytes()...)
 }
 
 //________________________________________________________________________________
 
-// get the key for an unbonding delegation
-func GetUBDKey(delegatorAddr, validatorAddr sdk.Address, cdc *wire.Codec) []byte {
-	return append(GetUBDsKey(delegatorAddr, cdc), validatorAddr.Bytes()...)
+// get the key for an unbonding delegation by delegator and validator addr.
+// VALUE: stake/types.UnbondingDelegation
+func GetUBDKey(delegatorAddr, validatorAddr sdk.Address) []byte {
+
+	key := make([]byte, len(delegatorAddr.Bytes()))
```

Comment:
i think we can actually do without this key and copy now that we have the order right in the append below

Reply:
you're correct - addressed